lib/makara/proxy.rb

```diff
@@ -59,22 +59,18 @@ def initialize(config)
     end
 
     def without_sticking
-      before_context = @master_context
-      @master_context = nil
       @skip_sticking = true
       yield
     ensure
       @skip_sticking = false
-      @master_context ||= before_context
     end
 
     def hijacked?
       @hijacked
     end
 
-    def stick_to_master!(write_to_cache = true)
-      @master_context = Makara::Context.get_current
-      Makara::Context.stick(@master_context, @id, @ttl) if write_to_cache
```

Comment:
This `write_to_cache` thing seems important, but it really would complicate your `Context` checks.
Now that I think about it, things get a little weird in the new model with really long requests. It seems against the expected behavior to `stick` in a request, it be a long request, and then unstick later in that request. Let's call this "Full Request Sticking" and I'll put a note above in the context.

Here, let's keep the `write_to_cache` variable and pass it to the `Context`.